app/experimenter/nimbus-ui/src/components/PageRequestReview/index.tsx

```diff
@@ -30,12 +30,31 @@ type PageRequestReviewProps = {
 const PageRequestReview = ({
   /* istanbul ignore next - only used in tests & stories */
   polling = true,
-}: PageRequestReviewProps) => {
+}: PageRequestReviewProps) => (
+  <AppLayoutWithExperiment
+    title="Review &amp; Launch"
+    testId="PageRequestReview"
+    {...{ polling }}
+    redirect={({ status }) => {
+      if (status.launched) {
+        // Return to the experiment root/summary page
+        return "";
+      }
+    }}
+  >
+    {({ experiment, refetch }) => <PageContent {...{ experiment, refetch }} />}
+  </AppLayoutWithExperiment>
+);
+
+const PageContent: React.FC<{
```

Comment:
I did this so we could stop using using refs to pass the experiment around in this component. Open to suggestions if this is too clunky. I wish we were using Context to pass experiment data around.

Reply:
Yeah, I've kind of got it in the back of my head & todo list to take a swing at moving this experiment lookup into Context. A couple wrinkles there that *could* be worked out maybe

That said, IMO this PageContent subcomponent is fine. It at least gets us out of a few levels of indentation

Reply:
@jodyheavener did you already have a WIP branch of using `Context` from a few months back or am I misremembering something? Would be out of date now but that sounds vaguely familiar.

Reply:
@LZoog @lmorchard yeah! I have [this](https://github.com/mozilla/experimenter/compare/jh/context?expand=1) (now very outdated) branch from a few months back where I moved experiment and analysis data to Context. Doesn't do much for LoC but it made things so much simpler IMO. One side effect I saw was that we could largely stop mocking the experiment GQL query in tests/Storybook and instead just mock the context value. It was too large to PR all at once so I was hoping to one day start incrementally moving towards it.